lib/spack/llnl/util/link_tree.py

```diff
@@ -29,6 +29,115 @@ def remove_link(src, dest):
         os.remove(dest)
 
 
+class MergeVisitor(object):
+    """
+    Visitor that produces actions:
+    - An ordered list of directories to create in dst
+    - An ordered list of files to link in dst
+    - A list of merge conflicts in dst/
+    """
+    def __init__(self, ignore=None):
+        self.ignore = ignore
```

Comment:
```suggestion
        self.ignore = ignore or lambda x: False
```
Then we can avoid having the `is_ignored` wrapper method and we can just call `self.ignore` directly.